ansible_runner/runner.py

```diff
@@ -145,13 +145,35 @@ def run(self):
             # If this is containerized, the shell environment calling podman has little
             # to do with the actual job environment, but still needs PATH, auth, etc.
             pexpect_env = os.environ.copy()
-            # But we still rely on env vars to pass secrets
-            pexpect_env.update(self.config.env)
             # Write the keys to pass into container to expected file in artifacts dir
             # option expecting should have already been written in ansible_runner.runner_config
-            env_file_host = os.path.join(self.config.artifact_dir, 'env.list')
+            env_file_host = os.path.join(self.config.private_data_dir, 'env', 'env.list')
             with open(env_file_host, 'w') as f:
-                f.write('\n'.join(list(self.config.env.keys())))
+                f.write('\n'.join(
+                     [f"{k}={v}" for k, v in self.config.env.items()]
+                ))
+
+            # This here is the magic sauce. The user inside of the container is in the root
+            # group, which is mapped to the host user's gid. By setting setgid on the artifacts
+            # dir + any subdirectories, any files created inside them will belong to the
+            # (in-container) root / host user's group, rather than the container user's gid.
+            # This allows runner's event_handler function to interact with these files outside
+            # the container.
+            if 'podman' in self.config.process_isolation_executable:
+                for root, dirs, files in os.walk(self.config.artifact_dir):
+                    dir_perms = (stat.S_IRUSR |
+                                 stat.S_IWUSR |
+                                 stat.S_IXUSR |
+                                 stat.S_IRGRP |
+                                 stat.S_IWGRP |
+                                 stat.S_IXGRP |
+                                 stat.S_ISGID)
+
+                    os.chmod(root, dir_perms)
```

Comment:
it would be cleaner to put this before the `os.walk` loop, unless I'm missing something.

Reply:
I don't think it really matters, but sure.